src/components/route-button-container.tsx

```diff
@@ -12,38 +19,39 @@ const styles = StyleSheet.create({
 	},
 });
 
-interface Props {
-	/** Function to start and stop route tracking */
-	changeTracking: () => void;
-	/** Function to toggle show route state true or false */
-	changeShowRoute: () => void;
-	/** State to determine whether route is visible or not */
-	showRoute: boolean;
-	/** State to determine whether route tracking has started or not */
-	isTracking: boolean;
-}
-
 /**
  * Route button container component which contains buttons to
  * start / end route tracking and to toggle route visibility on / off.
  * Just preliminary styling and location on a screen.
  * @returns a tree of React elements
  */
-const RouteButtonContainer = ({
-	changeTracking,
-	changeShowRoute,
-	showRoute,
-	isTracking,
-}: Props): JSX.Element => {
+const RouteButtonContainer = (): JSX.Element => {
+	const userId: string | null = useTypedSelector((state) => state.user.userId);
```

Comment:
If `state.user.userId` can be `null`, it should have that type in the `User` state in `src/reducers/user-reducer.ts`!

Reply:
Fixed